### PR TITLE
Implement Postman in pipelines leveraging Newman for command line triggering

### PR DESCRIPTION
**Pull request by RandallKeur**

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      - name: Build Docker image
+        run: docker build . --file Dockerfile --tag purrgent-care:$(date +%s)
+
+      - name: Start Spring Boot application
+        run: nohup mvn spring-boot:run & echo $! > spring-boot.pid
+
+      - name: Wait for Spring Boot to start
+        run: |
+          echo "Waiting for Spring Boot to start..."
+          timeout 60s bash -c 'until curl -s http://localhost:8080/actuator/health | grep "UP"; do sleep 2; done'
```
Comment (RandallKeur) [inline .github/workflows/ci-cd.yml:58]: The actuator below is needed to check whether the app is up and running before running the postman collection against the application.

File: pom.xml [hunk 2]
```diff
@@ -66,7 +66,10 @@
 			 <version>1.18.32</version>
 		</dependency>
 
-
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-actuator</artifactId>
+		</dependency>
```
Comment (RandallKeur) [inline pom.xml:72]: This dependency is part 1/2 to enable the actuator health check

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -11,4 +11,12 @@ spring:
...
\ No newline at end of file
+    init.mode: always
+  management:
+    endpoints:
+      web:
+        exposure:
+          include: health,info
+    endpoint:
+      health:
+        show-details: always
```
Comment (RandallKeur) [inline src/main/resources/application.yml:22]: This is part 2/2 to enable the actuator health check

File: makefile [hunk 4]
```diff
@@ -8,12 +13,16 @@ build-backend:
 test-backend:
 	mvn test jacoco:report
 
+test-postman:
+	newman run PurrgentCare.postman_collection.json --env-var "personsBaseURL=$(PERSONS_BASE_URL)" \
+	--env-var "petBaseURL=$(PET_BASE_URL)" --env-var "accountBaseURL=$(ACCOUNT_BASE_URL)"
+
```
Comment (RandallKeur) [inline makefile:19]: enabling postman collection to be run from the terminal directly through newman.

You might need to install newman for this to work...
`cd purrgent-care-ui & npm install -g newman`

File: .github/workflows/ci-cd.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        env:
+          MYSQL_ROOT_PASSWORD: root
+          MYSQL_DATABASE: testdb
+          MYSQL_USER: testuser
+          MYSQL_PASSWORD: testpassword
+        options: >-
+          --health-cmd="mysqladmin ping --silent"
+          --health-interval=10s
+          --health-timeout=5s
+          --health-retries=3
```
Comment (RandallKeur) [inline .github/workflows/ci-cd.yml:30]: configuring a mysql database to store our data for the application in the pipeline

File: .github/workflows/ci-cd.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+          node-version: '22'
+
+      - name: Install Newman
+        run: npm install -g newman
+
+      - name: Run Postman Collection
+        env:
+          PERSONS_BASE_URL: http://localhost:8080/persons
+          PET_BASE_URL: http://localhost:8080/pets
+          ACCOUNT_BASE_URL: http://localhost:8080/accounts
```
Comment (RandallKeur) [inline .github/workflows/ci-cd.yml:72]: These would be better as secrets in the repository but a repo admin needs to set those up.

Comment (RandallKeur) [reply .github/workflows/ci-cd.yml:72]: This will be addressed by this [issue](https://github.com/JesseMattson/PurrgentCare/issues/62) by @JesseMattson  😄

**Review by JesseMattson — APPROVED** [review]
make test-postman worked for me with this branch 👍

**Pull request merged**